gluten-celeborn/src/main/scala/org/apache/spark/shuffle/CelebornColumnarBatchSerializer.scala

```diff
@@ -61,21 +62,21 @@ private class CelebornColumnarBatchSerializerInstance(schema: StructType,
         .contextInstance()
         .newChildAllocator("GlutenColumnarBatch deserialize", 0, Long.MaxValue)
 
-      private val jniByteInputStream = JniByteInputStreams.create(in)
+      private lazy val jniByteInputStream = JniByteInputStreams.create(in)
+      private lazy val cSchema = ArrowSchema.allocateNew(ArrowBufferAllocators.contextInstance())
 
       private lazy val shuffleReaderHandle = {
-        val cSchema = ArrowSchema.allocateNew(ArrowBufferAllocators.contextInstance())
         val arrowSchema =
           SparkSchemaUtil.toArrowSchema(schema, SQLConf.get.sessionLocalTimeZone)
         GlutenArrowAbiUtil.exportSchema(allocator, arrowSchema, cSchema)
         val handle = ShuffleReaderJniWrapper.make(jniByteInputStream, cSchema.memoryAddress(),
           NativeMemoryAllocators.contextInstance.getNativeInstanceId)
-        cSchema.close()
         // Close shuffle reader instance as lately as the end of task processing,
         // since the native reader could hold a reference to memory pool that
         // was used to create all buffers read from shuffle reader. The pool
         // should keep alive before all buffers to finish consuming.
         TaskMemoryResources.addRecycler(50) { _ =>
+          close()
```

Comment:
Is this the case that the close condition `batchHandle == -1L` wasn't reached? How does it relate to speculative execution?

Reply:
Here are the questions about BatchSerializer fixed in this pr.
1. `cSchema` was not closed when the task was interrupted in `ShuffleReaderJniWrapper.make`;
2. `shuffleReaderHandle` was initialized while creating new `DeserializationStream`, the interrupt in `ShuffleReaderJniWrapper.make` was not catched, this is fixed in `GlutenColumnarBatchSerializer`;
3. After moving `cSchema.close` to `close`, it seems that `close` is not called after task is finished and left memory leak problem. it can be reproduce with removing `close` code and run unit `Test get_json_object datatab function`.